web/src/pm_list.js

```diff
@@ -162,13 +163,19 @@ export function handle_narrow_activated(filter) {
         scroll_pm_into_view($active_filter_li);
         update_private_messages();
     }
+    if (activity_ui.user_filter) {
+        activity_ui.build_user_sidebar();
+    }
 }
 
 export function handle_message_view_deactivated() {
     // Since one can renarrow via the keyboard shortcut or similar, we
     // avoid disturbing the zoomed state here.
     unhighlight_all_private_messages_view();
     update_private_messages();
+    if (activity_ui.user_filter) {
+        activity_ui.build_user_sidebar();
+    }
```

Comment:
You main import cycle bug is that you shouldn't be adding this code here. `pm_list` is the module for the left sidebar "direct messages" section; it has nothing to do with the right sidebar. You can call your live-updates directly from `web/src/views_util.js` and `handle_post_view_change`.

Reply:
Oh of course, thanks

Reply:
Maybe I had thought this was a different `handle_message_view_deactivated` function...? Sorry for not realizing this

Reply:
Okay I think I got it a lot better, but now I have `views_util > activity_ui > inbox_ui > views_util`